### DISCUSSION TITLE
More type of meal in meal planner and meal planner rule

### DISCUSSION THREAD

**Discussion by cavuminfundo** [mealie-recipes/mealie#3692] [Ideas]
Hello everyone,

First thanks to all the contributors of this projet. Mealie is great I love it!

I'm tryng to set up the meal planner for my diet but in have encountered some problem in doing it.

Some diet occour in more than just 3 type of meal ( breakfast - launch - dinner and sides), will be very useful to add some other type of meal in meal planner whit the possibilty to randomize them, like ''morning snack''  - '' afternon snack'' - and some integration ( like vitamines or others ); also very useful could be the possibilty to link ''sides'' to the main course (eg. ''launch sides'' and '' dinner sides''

Maybe a way to do so would be to leave the user the possibility to modify the type of meal and create others for a full customization of the meal rule planner.

An other key feature could be some ''dice that randomize all day'' - since you have prepared all the ruleset in the meal planner with one button mealie create an all day program with the rule you have planned before. 

Thanks all for your work!

**Comment by raseitz1011** [comment]
I would more options!  Such as morning snack, afternoon snack, evening snack like cavuminfundo said, would be great and allow me to guide/instruct which foods are good for snacking and which aren't.  Thanks for all the work you do, though.  The system is really great.

**Comment by ishirbhan** [comment]
More generally, it may be useful to allow for meal types to be customizable, including specifying an order. "Side" to me is not a meal, but rather a component of a meal, but allowing a customizable list of mealtypes would solve this. Specifying an order would be helpful, especially for folks using an API, since you won't know in advance what the mealtypes are.

**Comment by DaveChild** [comment]
This would be great. I usually batch-cook my lunches on a Sunday, so I'd like to add a "Meal-Prep" type to that list.

**Comment by richardgaywood** [comment]
On a related note - I am currently giving Mealie my first proper go by using it to plan a BBQ for some guests tomorrow. I would like a "prep" type, so I could add a recipe twice - once tomorrow, in the "lunch" slot, and once today, in the "prep" slot. This would keep me organised today as I prep some things ahead of time.

And then, for the BBQ tomorrow, I think I'd like an "Event" type, rather that lumping it into Lunch or Dinner with all my more mundane meals. It would be useful when looking back through the history to make these stand out more.

**Comment by Sail-Line** [comment]
It may be useful to allow for meal types to be customizable

**Comment by locnide** [comment]
It would also be great if the default meal type could be set. Currently dinner is the default but I use lunch most of the time.
